core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/request/RevisionIndexRequestBuilder.java

```diff
@@ -29,14 +30,32 @@
  * @since 5.7
  * @param <R> - the return type
  */
-public interface RevisionIndexRequestBuilder<R> extends RequestBuilder<BranchContext, R> {
+public interface RevisionIndexRequestBuilder<R> extends BranchRequestBuilder<R> {
 
+	@Override
 	default AsyncRequest<R> build(String repositoryId, String branch) {
-		return new AsyncRequest<>(
-			new RepositoryRequest<>(repositoryId,
-				new BranchRequest<>(branch, 
-					new RevisionIndexReadRequest<>(build())
+		// if the branch starts with MAIN, then it is an explicit branch path with a repositoryId
+		if (Strings.nullToEmpty(branch).startsWith(Branch.MAIN_PATH)) {
+			return new AsyncRequest<>(
+				new RepositoryRequest<>(repositoryId,
+					new HealthCheckingRequest<>(
+						new BranchRequest<>(branch, 
+							new RevisionIndexReadRequest<>(build())
+						),
+						allowedHealthstates()
+					)
 				)
+			);
+		} else {
+			return build(branch);
+		}
+	}
+	
+	default AsyncRequest<R> build(String codeSystemUri) {
+		return new AsyncRequest<>(
+			new CodeSystemResourceRequest<>(
+				codeSystemUri,
+				new RevisionIndexReadRequest<>(build())
```

Comment:
The URI-based alternative doesn't include the health check request wrapper.

Reply:
It does, internally.

Reply:
Well, I can add it to the execute method as well.